src/construction.cpp

```diff
@@ -1131,6 +1134,21 @@ bool construct::check_support( const tripoint &p )
     return num_supports >= 2;
 }
 
+bool construct::check_stable( const tripoint &p )
+{
+    return get_map().has_flag( flag_SUPPORTS_ROOF, p + tripoint_below );
+}
+
+bool construct::check_empty_stable( const tripoint &p )
+{
+    return check_empty( p ) && check_stable( p );
+}
+
+bool construct::check_open_air_above( const tripoint &p )
+{
+    return get_map().ter( p + tripoint_above ) == t_open_air;
```

Comment:
As I commented in your other PR, could you implement this without a hardcoded id check?

Reply:
Sure, I'll use NO_FLOOR flag

Reply:
Thank you!